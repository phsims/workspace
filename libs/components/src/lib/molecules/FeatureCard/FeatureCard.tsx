'use client';

import { useRouter } from 'next/navigation';
import Image from 'next/image';
import {
  Button,
  Card,
  Typography,
  CardContent,
  CardActions,
} from '@mui/material';
import ArrowForwardIosRoundedIcon from '@mui/icons-material/ArrowForwardIosRounded';

export interface FeatureCardProps {
  title: string;
  description: string;
  image: string;
  link: string;
}

export function FeatureCard({
  title,
  description,
  image,
  link,
}: FeatureCardProps) {
  const router = useRouter();

  return (
    <Card
      elevation={1}
      sx={{
        textAlign: 'center',
        display: 'flex',
        flexDirection: 'column',
        overflow: 'visible',
        mt: 6,
        position: 'relative',
      }}
    >
      <Image
        src={image}
        alt={title}
        height={256}
        width={256}
        style={{ margin: '-90px auto  -90px ' }}
      />
      <CardContent>
        <Typography variant="h5" sx={{ mb: 2, fontWeight: 600 }}>
          {title}
        </Typography>
        <Typography variant="body1" sx={{ mb: 0 }}>
          {description}
        </Typography>
      </CardContent>
      <CardActions sx={{ mt: 'auto', justifyContent: 'center' }}>
        <Button
          variant="text"
          onClick={() => router.push(link)}
          sx={{ fontSize: 'large', textTransform: 'none' }}
          endIcon={<ArrowForwardIosRoundedIcon fontSize="small" />}
        >
          Learn more
        </Button>
      </CardActions>
    </Card>
  );
}

export default FeatureCard;
