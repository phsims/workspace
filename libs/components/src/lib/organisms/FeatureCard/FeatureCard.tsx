import { useRouter } from 'next/navigation';
import Image from 'next/image';
import {
  Box,
  Button,
  Card,
  Typography,
  CardContent,
  CardActions,
} from '@mui/material';
import ArrowForwardIosRoundedIcon from '@mui/icons-material/ArrowForwardIosRounded';

export interface FeatureCardProps {
  title: string;
  description: string;
  image: string;
  link: string;
}

export function FeatureCard({
  title,
  description,
  image,
  link,
}: FeatureCardProps) {
  const router = useRouter();
  return (
    <Card
      elevation={1}
      sx={{
        textAlign: 'center',
        display: 'flex',
        flexDirection: 'column',
       
        overflow: 'visible',
        mt: 8,
        mb: 8,
        position: 'relative',
      }}
    >
      <Box sx={{ position: 'absolute', width: '100%', height: '270px' ,top:'-90px'}}>
        <Image
          src={image}
          alt={title}
          sizes="270px"
          fill
          style={{
            objectFit: 'cover',

          }}
        />
      </Box>

      {/* <Box
        component="img"
        src={image}
        alt={title}
        sx={{ width: '100%', height: 'auto', mt: '-35%', mb: '-40%' }}
      /> */}
      <CardContent sx={{mt:'70px'}}>
        <Typography variant="h5" sx={{ mb: 2, fontWeight: 600 }}>
          {title}
        </Typography>
        <Typography variant="body1" sx={{ mb: 0 }}>
          {description}
        </Typography>
      </CardContent>
      <CardActions sx={{ mt: 'auto', justifyContent: 'center' }}>
        <Button
          variant="text"
          onClick={() => router.push(link)}
          sx={{ fontSize: 'large', textTransform: 'none' }}
          endIcon={<ArrowForwardIosRoundedIcon fontSize="small" />}
        >
          Learn more
        </Button>
      </CardActions>
    </Card>
  );
}

export default FeatureCard;
