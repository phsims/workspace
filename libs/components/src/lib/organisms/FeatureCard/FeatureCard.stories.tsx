import type { Meta, StoryObj } from '@storybook/react';
import { FeatureCard } from './FeatureCard';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { Box } from '@mui/material';

const meta: Meta<typeof FeatureCard> = {
  component: FeatureCard,
  title: '@organisms/FeatureCard',
  decorators: [
    (Story) => (
      <Box sx={{width:345,mt:'5rem'}}>
        <Story />
      </Box>
    ),
  ],
};
export default meta;
type Story = StoryObj<typeof FeatureCard>;

export const Primary = {
  args: {
    title: "Shopping List Generator",
    description: "Automatically generate shopping lists from your meal plans.",
    image: "/featureFour.svg",
  },
};

export const Heading: Story = {
  args: {...Primary.args},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to FeatureCard!/gi)).toBeInTheDocument();
  },
};
