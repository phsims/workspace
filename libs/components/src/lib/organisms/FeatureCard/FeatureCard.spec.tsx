import { render } from '@testing-library/react';


import FeatureCard, { FeatureCardProps } from './FeatureCard';

  const defaultProps: FeatureCardProps = {
    title: 'Sample Title',
    description: 'This is a sample description.',
    image: 'https://via.placeholder.com/150',
    link: '/sample-link',
  };

  jest.mock('next/navigation', () => require('next-router-mock'));

describe('FeatureCard', () => {

  it('should match snapshot successfully', () => {
    const { baseElement } = render(<FeatureCard {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });
  it('renders with default props', () => {
    const { getByRole, getByText } = render(<FeatureCard {...defaultProps} />);

    const image = getByRole('img', { name: /sample title/i });
    const title = getByText(/sample title/i);
    const description = getByText(/this is a sample description./i);
    const button = getByRole('button', { name: /learn more/i });

    expect(image).toBeInTheDocument();
    expect(title).toBeInTheDocument();
    expect(description).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });
});
