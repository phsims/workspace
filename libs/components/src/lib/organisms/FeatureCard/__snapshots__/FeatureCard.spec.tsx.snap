// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureCard should match snapshot successfully 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rrzovd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-vkeyl2"
      >
        <img
          alt="Sample Title"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="270px"
          src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
      <div
        class="MuiCardContent-root css-1eox0ct-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1himrsc-MuiTypography-root"
        >
          Sample Title
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1wczmbf-MuiTypography-root"
        >
          This is a sample description.
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-13rhmik-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-azzldy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Learn more
          <span
            class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="ArrowForwardIosRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.38 21.01c.49.49 1.28.49 1.77 0l8.31-8.31c.39-.39.39-1.02 0-1.41L9.15 2.98c-.49-.49-1.28-.49-1.77 0s-.49 1.28 0 1.77L14.62 12l-7.25 7.25c-.48.48-.48 1.28.01 1.76"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
